src/internal/util/logger.ts

```diff
@@ -1,119 +1,286 @@
+         * Set format of console log
+         */
+        format?: LoggingFormat,
+        /**
+         * Redirect console.log methods to logger sub-system
+         */
+        redirect?: boolean,
+    };
+    file: {
+        /**
+         * Enable file logging
+         */
+        enabled: boolean,
+        /**
+         * Set log level for file logging
+         */
+        level?: string,
+        /**
+         * Set file name and path to log into
+         */
+        filename?: string,
+        /**
+         * Set format for file logging
+         */
+        format?: LoggingFormat,
+    };
+}
+
+/**
+ * Logging configuration to simply pass through log message to the console
+ */
+export const NoLoggingConfiguration: LoggingConfiguration = {
+    console: {
+        enabled: true,
+        level: "info",
+        format: LoggingFormat.None,
+        redirect: false,
+    },
+    file: {
+        enabled: false,
+    },
+};
+
+/**
+ * CLI-style logging configuration
+ */
+export const MinimalLoggingConfiguration: LoggingConfiguration = {
+    console: {
+        enabled: true,
+        level: "info",
+        format: LoggingFormat.Simple,
+        redirect: false,
+    },
+    file: {
+        enabled: false,
+    },
+};
 
-    winstonLogger.add(
-        winston.transports.File,
-        {
-            level,
+/**
+ * Default logging configuration for running automation clients
+ */
+export const DefaultClientLoggingConfiguration: LoggingConfiguration = {
+    console: {
+        enabled: true,
+        level: "info",
+        format: LoggingFormat.Full,
+        redirect: true,
+    },
+    file: {
+        enabled: false,
+    },
+};
+
+/**
+ * Configure the logging sub-system with the provided LoggingConfiguration
+ * It is safe to call this method several times to re-configure the logger.
+ * @param config
+ */
+export function configureLogging(config: LoggingConfiguration) {
+    winstonLogger.silent = true;
+    winstonLogger.clear();
+
+    // Set up console logging
+    if (config.console.enabled === true) {
+        const ct = new winston.transports.Console({
+            level: config.console.level || "info",
+            format: getFormat(config.console.format),
+        });
+        winstonLogger.add(ct);
+
+        if (config.console.redirect === true) {
+            redirectConsoleLogging();
+        } else {
+            unRedirectConsoleLogging();
+        }
+
+        winstonLogger.silent = false;
+    }
+
+    // Set up file logging
+    if (config.file.enabled) {
+
+        let filename = config.file.filename;
+
+        if (!filename) {
+            const appDir = __dirname.split(p.join("node_modules", "@atomist", "automation-client"))[0];
+            let pj: { name: string; };
+            try {
+                pj = require(p.join(appDir, "package.json"));
+            } catch (e) {
+                pj = { name: "atm-client" };
+            }
+            filename = p.join(
+                os.homedir(),
```

Comment:
I did not know that existed.

Note to self: use that when locating the user configuration file.